infra/blueprint-test/modules/tft/terraform.go

```diff
@@ -0,0 +1,248 @@
+		f.setupPath = setupPath
+	}
+}
+
+func WithVars(vars map[string]interface{}) option {
+	return func(f *TFBlueprintTest) {
+		f.vars = vars
+	}
+}
+
+func WithLogger(logger *logger.Logger) option {
+	return func(f *TFBlueprintTest) {
+		f.logger = logger
+	}
+}
+
+// Init sets defaults, validates and returns a TFBlueprintTest.
+func Init(t testing.TB, opts ...option) *TFBlueprintTest {
+	tft := &TFBlueprintTest{
+		name: fmt.Sprintf("%s TF Blueprint", t.Name()),
+		t:    t,
+	}
+	// apply options
+	for _, opt := range opts {
+		opt(tft)
+	}
+	// if no custom logger, set default based on test verbosity
+	if tft.logger == nil {
+		tft.logger = utils.GetLoggerFromT()
+	}
+	// one of fixture name or tfDir should be provided
+	if tft.fixtureName != "" && tft.tfDir != "" {
+		t.Fatalf("Only one of FixtureName or TFDir must be provided, found FixtureName=%s, TFDir=%s", tft.fixtureName, tft.tfDir)
+	}
+	// if both fixture name and tfDir are empty, auto discover tfDir based on cwd
+	if tft.fixtureName == "" && tft.tfDir == "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			t.Fatalf("unable to get wd :%v", err)
+		}
+		tfdir, err := discovery.ConfigDirFromCWD(cwd)
+		if err != nil {
+			t.Fatalf("unable to detect TFDir :%v", err)
+		}
+		tft.tfDir = tfdir
+	}
+	// if fixture name compute tfDir from given fixture name
+	if tft.fixtureName != "" {
+		tfModFixtureDir := getTFModuleFixtureDir(tft.fixtureName)
+		if _, err := os.Stat(tfModFixtureDir); os.IsNotExist(err) {
+			t.Fatalf("TFDir path derived from %s as %s does not exist", tft.fixtureName, tfModFixtureDir)
+		}
+		tft.tfDir = tfModFixtureDir
+	}
+	// setupPath is empty, try known setupPath
+	if tft.setupPath == "" {
+		tfModSetupDir := getTFModuleSetupDir()
+		if _, err := os.Stat(tfModSetupDir); os.IsNotExist(err) {
+			t.Logf("Setup dir %s not found, skipping loading setup outputs as fixture inputs", tfModSetupDir)
+		}
+		tft.setupPath = tfModSetupDir
+	}
+	// load TFEnvVars
+	tfEnvVars := make(map[string]string)
+	//load from setup outputs
+	if tft.setupPath != "" {
+		tft.logger.Logf(tft.t, "Loading env vars from setup %s", tft.setupPath)
+		loadTFEnvVar(tfEnvVars, tft.getTFSetupOPMap())
+	}
+	// load additional env variables
+	if tft.envVars != nil {
+		loadTFEnvVar(tfEnvVars, tft.envVars)
+	}
+	tft.envVars = tfEnvVars
+
+	tft.logger.Logf(tft.t, "Running tests TF configs in %s", tft.tfDir)
+	return tft
+}
+
+// getTFModuleFixtureDir gets TF fixture config path from a fixture name.
+func getTFModuleFixtureDir(fixture string) string {
+	return fmt.Sprintf("../../test/fixtures/%s", fixture)
+}
+
+// getTFModuleSetupDir returns well known setup path.
+func getTFModuleSetupDir() string {
+	return "../../setup"
+}
+
+// getTFOptions generates terraform.Options used by Terratest.
+func (b *TFBlueprintTest) getTFOptions() *terraform.Options {
+	return terraform.WithDefaultRetryableErrors(b.t, &terraform.Options{
+		TerraformDir: b.tfDir,
+		EnvVars:      b.envVars,
```

Comment:
Is this meant to be prefixed EnvVars or the unprefixed ones? (This is confusing from earlier.)

Reply:
This is supposed to be the prefixed envvars. I have renamed to `tfEnvVars` and updated struct comments to make this clearer.